### PR TITLE
DISC-800/Refactoring of Search.vue.

### PR DESCRIPTION
**Pull request by jesperlauridsen**
There we go. Lotta refactoring, I think it's much nicer now to look at, all split up.I tried to move stuff where it belongs, and had to move the variable for showing the facets into the store, because we have several components listening for that change to make it work nicely.

Have a loot at it, and see if you think its okey - or if you find that i broke anything!

### CODE REVIEW

**Review by jorntx — COMMENTED** [review]
Very nice refactor - cleaned up very well! Only minor stuff.

Took it for a spin and seems to work as it did before.

File: src/components/common/PortalContent.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			</div>
+		</div>
+	</div>
+</template>
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { useSearchResultStore } from '@/store/searchResultStore';
+import GridDisplay from '@/components/common/GridDisplay.vue';
+//The import is actually used so had to silence TypeScript here - have no idea why compiler is angry
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/common/PortalContent.vue:46]: You actually put that in there, a few months ago, according to git blame :D

[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/common/PortalContent.vue:46]: I think our lint re-setup took care of it. I'll remove it.

File: src/components/search/SearchOverhead.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<template>
+	<div class="container">
+		<div class="row">
+			<!-- If there was ANY this to show -->
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/SearchOverhead.vue:4]: Should be removed :D

File: src/components/search/SearchOverhead.vue
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/SearchOverhead.vue:1]: I really couldn't come up with a better name. It's the overhead of the search experience, showing sort, etc. If you got anything better, feel free to throw it at me. I got no feelings for the name. ;-)

File: src/components/search/Facets.vue [hunk 3]
```diff
@@ -103,15 +118,50 @@ export default defineComponent({
 			window.removeEventListener('filter-update', filterUpdateHelper);
 		});
 
+		/* This is because Vue3 composition API has this weird bug that when a ref is wrapped in a v-if
+		   the ref is not actually present in onMounted, which is super weird. But we can watch for when it enters.
+		   https://github.com/vuejs/composition-api/issues/296
+		*/
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/search/Facets.vue]: Reworked it now. It seems to work nicely now.

File: src/components/common/PortalContent.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorntx) [reply src/components/common/PortalContent.vue]: Getting old here - memory slipping:)

File: src/components/search/SearchOverhead.vue
[the remark this answers is not in the document]
Comment (jorntx) [reply src/components/search/SearchOverhead.vue:1]: I will rename if I come up with something better.

**Review by jorntx — APPROVED** [review]
We are good to go! Thanks for the changes.

**Pull request merged**